### ISSUE TITLE
Postcss

### ISSUE THREAD

**Issue by stylesuxx** [newtriks/generator-react-webpack#163]
I hope I do not annoy you with my CSS related requests.

I was wondering what you think about adding [postcss](https://github.com/postcss/postcss) support?
They have a massive amount of different plugins to post process your CSS and thus make your life easier - I especially like [postcss-grid](https://github.com/andyjansson/postcss-grid) and [postcss-autoprefixer](https://github.com/postcss/autoprefixer).

When running the generator one may be prompted if one wants to use **postcss** and then one may select to enable different plugins from different sections(as listed in the [postcss README](https://github.com/postcss/postcss)).

Again, as with #162, I would gladly submit a PR if this is a feature you would like to see.

**Comment by weblogixx** [comment]
Hi @stylesuxx,

no, you dont annoy me :). In fact, I am happy for your contributions. As stated in https://github.com/newtriks/generator-react-webpack/issues/162, I would provide the basics for postcss, but wont alter the generator to include the wanted plugins. People would have a chance to use it and include plugins if they need them. I dont want to add too much functionality in the base generator.

**Comment by stylesuxx** [comment]
OK, so this is related to [react-webpack templat PR 12](https://github.com/weblogixx/react-webpack-template/pull/12) also to #162, which referenced to the other PR :-)

You can see my *generator only* implementation of the **postcss** functionality in [my fork](https://github.com/newtriks/generator-react-webpack/compare/master...stylesuxx:feature/postcss).

As you can see, this *relies heavily on parsing* the base config after it has been copied to the project folder and adding/replacing the postcss relevant stuff - I *do not like this* too much, but right now I can not think of any other version without having to change the template.

Only have tested this with the "plain" css files yet, but it should work with sass and the others too.
Just return something like this from the *postcss* function in *cfg/base.js*:

``` JavaScript
return [
  require('autoprefixer')({
    browsers: ['last 2 versions', 'ie >= 8']
  })
];
```

And add some fancy style to App.js:

```CSS
body {
  hyphen: auto;
}
```

Make sure to install the **autoprefixer** package in your project.

@weblogixx what do you think? Any ideas how to solve this in a "cleaner" way?

From my point of view it would be cleaner to all have this in the template, but this would introduce the **postcss** and **postcss-loader** dev dependencies.

**Comment by weblogixx** [comment]
@stylesuxx: Seems good so far. Mind adding some tests for it? I know it is not too nice to adjust the config via regex, but I dont seem to be able to find a better way, too. I will check it out some more.

**Comment by stylesuxx** [comment]
Of course I will add some tests, just wanted to see if you are OK with this solution, otherwise I would just have trashed the branch and thought up something else ;-)

Which I did now anyhow, after talking to @sthzg a bit, who suggested to load *cfg/base.js* into an AST then traverse it and add/replace values where needed.

I [updated my fork](https://github.com/newtriks/generator-react-webpack/compare/master...stylesuxx:feature/postcss) accordingly.

I think this is *much cleaner*, since we do not need to maintain the regex'. Although it comes with a price of three extra dependencies when enabling postcss and some more code.

**Comment by weblogixx** [comment]
@stylesuxx: Honestly I first thought that it could be overkill, too. That was until I looked at it deeper. You are right that we might need the ast parser in the future anyway, so the dependencies arent that much of a problem.

Two final points:
- What do you think about moving ```this.postcssConfig()``` to its own file so we could require it? This would have the benefit that ```app/index.js``` would stay smaller (and, if no longer required, the changes could be more easily reverted).
- Please also add some notes of this feature in ```README.md``` so people know what it does:). 

Otherwise, I will happily merge when you provide the tested pr. Thats a very nice work, thank you!

**Comment by stylesuxx** [comment]
@weblogixx about the tests, should I just copy the *react-webpack:app* testcase and change the prompt for *PostCSS* accordingly, adding testcases to check the content of the config file - or what would you suggest?

**Comment by weblogixx** [comment]
@stylesuxx, yes I think that would be enough for the start. I also plan to also add some basic tests for required packages and such stuff in package.json, so I could built upon your pr.

**Comment by weblogixx** [comment]
Included and will be released as 3.1.0. Thanks again @stylesuxx!
